tests/monitors/kubernetes_cluster/perf_test.py

```diff
@@ -225,3 +225,178 @@ def missing_service_tags():
+                            "spec": {},
+                        },
+                        namespace="default",
+                    )
+
+            def has_all_deployment_props():
+                for _, rs in replicaSets.items():
+                    for pod_uid in rs["pod_uids"]:
+                        if not has_dim_prop(
+                            backend,
+                            dim_name="kubernetes_pod_uid",
+                            dim_value=pod_uid,
+                            prop_name="deployment",
+                            prop_value=rs["dp_name"],
+                        ):
+                            return False
+                        if not has_dim_prop(
+                            backend,
+                            dim_name="kubernetes_pod_uid",
+                            dim_value=pod_uid,
+                            prop_name="deployment_uid",
+                            prop_value=rs["dp_uid"],
+                        ):
+                            return False
+                    return True
+
+            def has_all_replicaSet_props():
+                for _, rs in replicaSets.items():
+                    for pod_uid in rs["pod_uids"]:
+                        if not has_dim_prop(
+                            backend,
+                            dim_name="kubernetes_pod_uid",
+                            dim_value=pod_uid,
+                            prop_name="replicaSet",
+                            prop_value=rs["rs_name"],
+                        ):
+                            return False
+                        if not has_dim_prop(
+                            backend,
+                            dim_name="kubernetes_pod_uid",
+                            dim_value=pod_uid,
+                            prop_name="replicaSet_uid",
+                            prop_value=rs["rs_uid"],
+                        ):
+                            return False
+                    return True
+
+            assert wait_for(p(has_datapoint, backend, dimensions={"kubernetes_pod_name": "pod-dp-0-replicaset-0"}))
+            assert wait_for(p(has_datapoint, backend, dimensions={"kubernetes_pod_name": "pod-dp-49-replicaset-99"}))
+
+            assert wait_for(has_all_deployment_props, interval_seconds=2)
+
+            for _, rs in replicaSets.items():
+                v1beta1_client.delete_namespaced_replica_set(name=rs["rs_name"], namespace="default", body={})
+                for pod_name in rs["pod_names"]:
+                    v1_client.delete_namespaced_pod(name=pod_name, namespace="default", body={})
+
+            time.sleep(30)
+
+            pprof_client.save_goroutines()
+
+            assert (
+                backend.datapoints_by_metric["sfxagent.go_num_goroutine"][-1].value.intValue < 100
+            ), "too many goroutines"
+
+            assert (
+                backend.datapoints_by_metric["sfxagent.go_heap_alloc"][-1].value.intValue < 2 * heap_usage_baseline
```

Comment:
Twice the baseline seems a bit high since technically everything that was put into heap for the monitor should be removed right?

Reply:
updated the test to await for heap 1.25x less than baseline